plugin/src/sbt-test/sbtix/private-auth/sbtix.nix

```diff
@@ -1,39 +1,53 @@
 { runCommand, fetchurl, lib, stdenv, jdk, sbt, writeText }:
+with stdenv.lib;
 rec {
-    mkMavenRepo = name: repo: runCommand name {}
+    mkRepo = name: artifacts: runCommand name {}
         (let
-            slashify = builtins.replaceStrings ["."] ["/"];
-            linkArtifact = organization: module: version: type: urlAttrs:
-                "ln -fsn ${fetchurl urlAttrs} $out/${slashify organization}/${module}/${version}/${module}-${version}.${type}";
-            linkVersion = organization: module: version: versionAttrs:
-                [ "mkdir -p $out/${slashify organization}/${module}/${version}/" ]
-                ++ lib.mapAttrsToList (linkArtifact organization module version) versionAttrs;
-            linkModule = organization: module: moduleAttrs:
-                lib.concatLists (lib.mapAttrsToList (linkVersion organization module) moduleAttrs);
-            linkOrganization = organization: organizationAttrs:
-                lib.concatLists (lib.mapAttrsToList (linkModule organization) organizationAttrs);
-            linkProject = projectName: project:
-                lib.concatLists (lib.mapAttrsToList linkOrganization project);
+            parentDirs = filePath: 
+                concatStringsSep "/" (init (splitString "/" filePath));
+            linkArtifact = outputPath: urlAttrs:
+                [ "mkdir -p \"$out/${parentDirs outputPath}\""
+                  "ln -fsn \"${fetchurl urlAttrs}\" \"$out/${outputPath}\""
+                ];
         in
-            lib.concatStringsSep "\n" (lib.concatLists (lib.mapAttrsToList linkProject repo)));
+            lib.concatStringsSep "\n" (lib.concatLists (lib.mapAttrsToList linkArtifact artifacts)));
+    
+    repoConfig = repos: nixrepo:
+    let
+        repoPatternOptional = repoPattern:
+            optionalString (repoPattern != "") ", ${repoPattern}";
+        repoPath = repoName: repoPattern:
+            [ "${repoName}: file://${nixrepo}/${repoName}${repoPatternOptional repoPattern}" ];
+    in
+        lib.concatStringsSep "\n  " (lib.concatLists (lib.mapAttrsToList repoPath repos));
 
-    buildSbtProject = args@{repo, name, buildInputs ? [], sbtOptions ? "", ...}:
-        stdenv.mkDerivation (rec {
-            mvn = mkMavenRepo "${name}-repo" repo;
+    mergeAttr = attr: repo:
+        fold (a: b: a // b) {} (catAttrs attr repo);
 
-            #eventually this repo list should only contain the nix repo
+    loadRepo = repo:
```

Comment:
I suppose that would let them supply additional dependencies based on options from their build. But the default.nix won't be as pretty.   `/etc/nixos/configuration.nix` does its importing with a list of paths usually `imports =[...]`.  I don't know if I have seen it anywhere else.

Also manual-repo could move back to default.nix.  I had moved it when I tried to deduplicate it and sbtix.nix, which didn't work out.